problem-three.js

```diff
@@ -0,0 +1,28 @@
+/*
+    Write a javascript function that accepts two arrays and merges them to produce an interleaved output
+
+    E.g
+
+    var A = [1, 3, 5, 7, ‘c’]
+    var B = [‘a’, 4, 6, ‘b’, 10, 12, 14]
+
+    Output  = [1, ‘a’, 3, 4, 5, 6, 7, ‘b’, ‘c’, 10, 12, 14]
+*/
+
+const mergeArrays = function (array1, array2){
+    let resArray = [];
+    
+    const arrayLength = array1.length > array2.length ? array1.length : array2.length
```

Comment:
So what you want to do here is slice your bigger array at the index of length of smaller one, then you run the loop with the first part of the bigger array, that way you will not need the checks. Then finally you concatenate the last part of your bigger array with the result. This way you don't run unnecessary iterations for the bigger array